src/cargo/util/lock.rs

```diff
@@ -0,0 +1,92 @@
+use file_lock::filename::{Mode, ParseError};
+use file_lock::filename::Lock as FileLock;
+use file_lock::filename::Error as FileLockError;
+
+use std::path::PathBuf;
+use std::fs;
+use std::thread::sleep_ms;
+
+use util::{Config, CargoError, CargoResult, human, caused_human};
+
+pub use file_lock::filename::Kind as LockKind;
+
+impl From<FileLockError> for Box<CargoError> {
+    fn from(t: FileLockError) -> Self {
+        Box::new(t)
+    }
+}
+
+impl From<ParseError> for Box<CargoError> {
+    fn from(t: ParseError) -> Self {
+        Box::new(t)
+    }
+}
+
+impl CargoError for FileLockError {
+    fn is_human(&self) -> bool { true }
+}
+impl CargoError for ParseError {}
+
+pub struct CargoLock {
+    kind: LockKind, 
+    inner: FileLock,
+}
+
+impl CargoLock {
+
+    pub fn lock_kind(config: &Config) -> CargoResult<LockKind> {
+        // TODO(ST): rename config key to something more suitable
+        const CONFIG_KEY: &'static str = "build.lock-kind";
```

Comment:
Do you mean cargo should just always abort, or always wait, without an option to configure it ?